austrakka/components/project/dataset/funcs.py

```diff
@@ -0,0 +1,127 @@
+                                                         file_hash=file_hash,
+                                                         custom_headers=custom_headers)
+    logger.info('Acknowledging...')
+    path_ack = "/".join([PROJECT_PATH, abbrev, DATASET_ACK_PATH, tracking_token])
+    return api_post(
+        path=path_ack,
+    )
+
+
+@logger_wraps()
+def track_dataset(
+        abbrev:str,
+        tracking_token: str,
+        detailed: bool,
+        out_format: str):
+    path = "/".join([PROJECT_PATH,
+                     abbrev,
+                     DATASET_TRACK_DETAILED_PATH if detailed else DATASET_TRACK_PATH,
+                     tracking_token])
+    if detailed:
+        response = api_get(path)
+        data = response['data'] if ('data' in response) else response
+        if not data:
+            logger.info("No JobFeedbacks available")
+            return
+
+        result = pd.DataFrame.from_dict(data)
+        print_table(
+            result,
+            out_format,
+        )
+    else:
+        call_get_and_print_dataset_status(
+            path,
+            out_format
+        )
+
+
+@logger_wraps()
+def add_dataset_blocking(
+        filepath: str,
+        label: str,
+        abbrev: str,
+        out_format: str):
+    logger.info('Storing')
+    path_adding = "/".join([PROJECT_PATH, abbrev, DATASET_UPLOAD_PATH])
+    filename = os.path.basename(filepath)
+
+    custom_headers = {
+        'analysis-label': label,
+        'filename': filename,
+    }
+
+    file_hash = get_hash(filepath)
+    with open(filepath, 'rb') as file_content:
+        files = [('files[]', (filename, file_content))]
+        tracking_token = upload_multipart_tracking_token(path=path_adding,
+                                                         files=files,
+                                                         file_hash=file_hash,
+                                                         custom_headers=custom_headers)
+    path_ack = "/".join([PROJECT_PATH, abbrev, DATASET_ACK_PATH, tracking_token])
+    api_post(
+        path=path_ack,
+    )
+    path_track = "/".join([PROJECT_PATH,
+                           abbrev,
+                           DATASET_TRACK_PATH,
+                           tracking_token])
+
+    while True:
+        logger.info('Tracking...')
+        status_change = call_get_and_print_table_on_state_change(
+            path_track,
+            out_format,
+            'Acknowledged',
+        )  # Replace this with your actual function to fetch status
+
+        if status_change is not None:
+            logger.success(f"Current status: {status_change}")
+
+            if status_change == 'Finished':
+                logger.success('')
+                break  # Exit the loop when the desired status is reached
+            if "Failed" in status_change:
+                logger.error("The Job in-jest has failed")
+                break
```

Comment:
Can we change error to "The dataset ingest has failed"? (Notice also typo in ingest)